### PR TITLE
kola/tests/misc/nfs: add RHCOS specific NFS test

### PR DESCRIPTION
**Pull request by arithx**
Updates the NFS test to no longer write to use /var/mnt rather than
/mnt, adds the ability to specify what is being mounted, and adds a
RHCOS specific NFSv4 test.

### CODE REVIEW

File: kola/tests/misc/nfs.go [hunk 1]
```diff
@@ -123,15 +131,20 @@ systemd:
...
 }
 
 // Test that NFSv4 without security works on CoreOS.
 func NFSv4(c cluster.TestCluster) {
-	testNFS(c, 4)
+	testNFS(c, 4, "/tmp")
+}
+
+// Test that NFSv4 without security works on RHCOS.
+func NFSv4(c cluster.TestCluster) {
```
Comment (miabbott) [inline kola/tests/misc/nfs.go]: Travis caught this...should be `RHCOSNFSv4`

Comment (arithx) [reply kola/tests/misc/nfs.go]: Yup, that's what I get for doing a rework right before the end of the day.

**Comment by miabbott** [comment]
The NFS specific commit looks sane to me.  I'll leave any comments about the distro flag in #874

File: kola/tests/misc/nfs.go [hunk 2]
```diff
@@ -92,13 +100,13 @@ systemd:
         Requires=rpc-statd.service
 
         [Mount]
-        What=%s:/tmp
-        Where=/mnt
+        What=%s:%s
+        Where=/var/mnt
```
Comment (cgwalters) [inline kola/tests/misc/nfs.go]: Related:  https://github.com/systemd/systemd/pull/6293

File: kola/tests/misc/nfs.go [hunk 3]
```diff
@@ -110,9 +118,9 @@ systemd:
 	c.Log("NFS client booted.")
 
 	checkmount := func() error {
-		status, err := c.SSH(m2, "systemctl is-active mnt.mount")
+		status, err := c.SSH(m2, "systemctl is-active var-mnt.mount")
```
Comment (cgwalters) [inline kola/tests/misc/nfs.go]: Shouldn't this use `mountLocation`?

Comment (arithx) [reply kola/tests/misc/nfs.go]: `mountLocation` might be a mis-leading variable name. It's intended to be what you're mounting on the remote system (in the service the `What` for the mount is defined as `<PrivateIP>:<mountLocation>`, the `Where` is always `/var/mnt`). No matter it's value the systemd mount will be `var-mnt.mount`.

File: kola/tests/misc/nfs.go [hunk 4]
```diff
@@ -49,15 +49,23 @@ func init() {
...
+	register.Register(&register.Test{
+		Run:         RHCOSNFSv4,
+		ClusterSize: 0,
+		Name:        "rhcos.linux.nfs.v4",
+		Distros:     []string{"rhcos"},
 	})
 }
 
-func testNFS(c cluster.TestCluster, nfsversion int) {
+func testNFS(c cluster.TestCluster, nfsversion int, mountLocation string) {
```
Comment (bgilbert) [inline kola/tests/misc/nfs.go]: `remotePath`?

File: kola/tests/misc/nfs.go [hunk 5]
```diff
@@ -123,15 +131,20 @@ systemd:
...
 // Test that the kernel NFS server and client work within CoreOS.
 func NFSv3(c cluster.TestCluster) {
-	testNFS(c, 3)
+	testNFS(c, 3, "/tmp")
 }
 
 // Test that NFSv4 without security works on CoreOS.
 func NFSv4(c cluster.TestCluster) {
-	testNFS(c, 4)
+	testNFS(c, 4, "/tmp")
```
Comment (bgilbert) [inline kola/tests/misc/nfs.go]: Any reason not to just switch to `/`?

Comment (arithx) [reply kola/tests/misc/nfs.go]: I'll test again tomorrow morning but if I recall correctly that did not work on CL.

**Comment by arithx** [comment]
Pushed with the variable name change and a fixed commit message.

File: kola/tests/misc/nfs.go (hunk 5 above)
Comment (bgilbert) [reply kola/tests/misc/nfs.go]: Nah, I wouldn't spend time on it.

**Comment by arithx** [comment]
Will rebase & merge once #874 lands

**Pull request merged**